src/compiler/output-targets/dist-custom-elements/custom-elements-build-conditionals.ts

```diff
@@ -19,7 +19,7 @@ export const getCustomElementsBuildConditionals = (
   const build = getBuildFeatures(cmps) as d.BuildConditionals;
 
   build.lazyLoad = false;
-  build.hydrateClientSide = false;
+  build.hydrateClientSide = true;
```

Comment:
Instead of always setting this value to `true`, which can cause Stencil to ship unnecessary code related to hydration even though the user is not intending to use this feature, I suggest to make this value depending on whether a `dist-hydrate-script` output target is set.

Similar to how we approach this in [`dist`](https://github.com/ionic-team/stencil/blob/9d2d0a0612caafb88c559907c6dec1a39e760615/src/compiler/output-targets/dist-lazy/lazy-build-conditionals.ts):

```ts
import { isOutputTargetHydrate } from '@utils';

const hasHydrateOutputTargets = config.outputTargets.some(isOutputTargetHydrate);
build.hydrateClientSide = hasHydrateOutputTargets;
```

Reply:
Good points, this way we do not introduce unattended changes for the people who are not using SSR. Implemented your suggestion and also added tests for it 👍